src/main/java/world/bentobox/bentobox/database/objects/Island.java

```diff
@@ -940,4 +940,33 @@ public String getGameMode() {
     public void setGameMode(String gameMode) {
         this.gameMode = gameMode;
     }
-}
\ No newline at end of file
+    
+    /**
+     * Returns if this island has its end island generated.
+     * @return {@code true} if this island has its end island generated, {@code false} otherwise.
+     * @since 1.5.0
+     */    
+    public boolean hasEndIsland(){
+        if (!BentoBox.getInstance().getIWM().isEndGenerate(getWorld()) || !BentoBox.getInstance().getIWM().isEndIslands(getWorld()) || BentoBox.getInstance().getIWM().getEndWorld(getWorld()) == null ){
+            return false;   
+        }
+        Location l = this.getCenter();
+        l.setWorld(BentoBox.getInstance().getIWM().getEndWorld(l.getWorld()));
+        return l.getBlock().getType() == Material.BEDROCK;
+    }
+
+    /**
+     * Returns if this island has its nether island generated.
+     * @return {@code true} if this island has its nether island generated, {@code false} otherwise.
+     * @since 1.5.0
+     */     
+    public boolean hasNetherIsland(){
+        if (!BentoBox.getInstance().getIWM().isNetherGenerate(getWorld()) || !BentoBox.getInstance().getIWM().isNetherIslands(getWorld()) || BentoBox.getInstance().getIWM().getNetherWorld(getWorld()) == null ){
```

Comment:
BetnoBox.getInstance().getIWM().isIslandNether(getWorld()).

Reply:
```
IslandWorldManager iwm = BentoBox.getInstance().getIWM();
return iwm.isNetherGenerate(getWorld()) &&
iwm.isNetherIslands(getWorld()) &&
iwm.getNetherWorld(getWorld()) != null &&
getCenter().toVector().toLocation(iwm.getNetherWorld(getWorld)).getBlock().getType().equals(Material.BEDROCK);
```

Reply:
^^^ I just wrote that from memory, so I might not have all the methods correct, but it avoids getting the island location and setting the world (I'm concerned the island center is mutable).